app/app/Http/Controllers/ChangePasswordController.php

```diff
@@ -32,36 +33,26 @@ public function post(Request $request)
         if (BaseUser::isSignedIn())
         {
 			$user = BaseUser::getDbUser();
-			$validation_rules = array(
-				'current_password'     => 'required',
-				'new_password'         => 'required',
-				'password_confirm'     => 'required|same:new_password'
-			);
-			$validator = Validator::make(Input::all(), $validation_rules);
-			$failing = $validator->fails();
-			if (!$failing && !empty($user->password_hash))
-			{
-				if ( !BaseUser::authenticate($user->email, $request->input('current_password')) )
-				{
-					$validator->errors()->add('current_password', 'Current password not matched');
-					$failing = true;
-				}
+			$validator = Validator::make(
+			    $request->all(), [
+			        'current_password' => 'required',
+                    'new_password' => 'required',
```

Comment:
Fix the indentation here.  It is really messed up.  current_password and new_password should line up.

This could be fixed by just undoing your changes to the $validation_rules variable.

Check your editor's tab and space preferences.

Reply:
Don't know why it showed like this here but in my editor it is showing perfect.